dropwizard-index-page/src/main/java/com/palantir/indexpage/FileSystemAsset.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.indexpage;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.github.mustachejava.DefaultMustacheFactory;
+import com.github.mustachejava.Mustache;
+import com.google.common.base.Charsets;
+import com.google.common.io.Files;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+
+/**
+ * Used to represent assets loaded from the file system.
+ */
+public final class FileSystemAsset implements Asset {
+
+    private final DefaultMustacheFactory factory;
+    private final File file;
+
+    private long lastModified;
+    private Mustache mustacheTemplate;
+
+    public FileSystemAsset(File file) {
+        checkNotNull(file);
+
+        this.factory = new DefaultMustacheFactory();
+        this.file = file;
+        this.lastModified = System.currentTimeMillis();
+    }
+
+    @Override
+    public Mustache getTemplate() {
+        try {
+            checkAndReloadFile();
+        } catch (FileNotFoundException e) {
+            return null;
```

Comment:
This is specific to the file system path as classpath path will never throw the exception